pkg/controllers/management/auth/roletemplates/common.go

```diff
@@ -140,24 +140,24 @@ func deleteClusterMembershipBinding(rtb metav1.Object, crbController crbacv1.Clu
 }
 
 // getMembershipRoleName returns the name of the membership role based on the RoleTemplate.
-func getClusterMembershipRoleName(rt *v3.RoleTemplate, rtb metav1.Object) string {
+func getClusterMembershipRoleName(rtb metav1.Object, isClusterOwner bool) string {
 	var clusterName string
 	switch obj := rtb.(type) {
 	case *v3.ProjectRoleTemplateBinding:
 		clusterName, _ = rbac.GetClusterAndProjectNameFromPRTB(obj)
 	case *v3.ClusterRoleTemplateBinding:
 		clusterName = obj.ClusterName
 	}
-	if isClusterOwnerRole(rt) {
+	if isClusterOwner {
 		return name.SafeConcatName(clusterName, clusterContext+"owner")
 	} else {
```

Comment:
You don't need `else` here...if the first branch is `true` you've already returned :-)